gocourse08/webinar/main.go

```diff
@@ -0,0 +1,50 @@
+/*
+написати генерік функції:
+створення посилання на простий тип PtrOf() (int → *int)
+
+	пошук у слайсі елементу FindIn()
+
+видалення зайвих елементів у слайсі з метою залишити лише унікальні Unique()
+
+	func PtrOf(v int) *int {
+	    return &v
+	}
+*/
+package main
+
+import "fmt"
+
+func PtrOf[T int | float64](v T) *T {
+	return &v
+}
+
+func FindIn[T comparable](sl []T, find T) int {
+	for i, v := range sl {
+		if v == find {
+			return i
+		}
+	}
+	return -1
+}
+
+func Unique[T comparable](sl []T) []T {
+	array := make(map[T]struct{})
```

Comment:
Можна трішки покращити назви:
```suggestion
	set := make(map[T]struct{})
```